crates/red_knot_python_semantic/resources/mdtest/functions.md

```diff
@@ -0,0 +1,176 @@
+# Functions
+
+## Call Expression
+
+### Simple
+
+Calling a simple function that returns an integer should infer the correct return type.
+
+```py
+def get_int() -> int:
+    return 42
+
+x = get_int()
+reveal_type(x)  # revealed: int
+```
+
+### TODO: Async
+
+NOTE: This is a TODO because we don't yet support `types.CoroutineType`. `x` should be generic `Coroutine[Any, Any, int]`!
+
+```py
+async def get_int_async() -> int:
+    return 42
+
+x = get_int_async()
+reveal_type(x)  # revealed: @Todo
+```
+
+### TODO: Decorated
+
+A function decorated with a decorator that alters its return type should reflect the decorated type.
+
+TODO: `x` should reveal `int`, as the decorator replaces `bar` with `foo`!
+
+```py
+from typing import Callable
+
+def foo() -> int:
+    return 42
+
+def decorator(func) -> Callable[[], int]:
+    return foo
+
+@decorator
+def bar() -> str:
+    return 'bar'
+
+x = bar()
+reveal_type(x)  # revealed: @Todo
+```
+
+### Constructor
```

Comment:
This doesn't belong under Functions heading; let's put it in `call/constructor.md`